pkg/cli/config/config.go

```diff
@@ -58,19 +60,30 @@ func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
 		}
 	}
 
-	// Check if user has explicitly defined backplane URL; it has higher precedence over the config file
-	err = viper.BindEnv("url", info.BackplaneURLEnvName)
+	// Check if user has explicitly defined proxy; it has higher precedence over the config file
+	err = viper.BindEnv("proxy-url", info.BackplaneProxyEnvName)
 	if err != nil {
 		return bpConfig, err
 	}
 
-	// Check if user has explicitly defined proxy; it has higher precedence over the config file
-	err = viper.BindEnv("proxy-url", info.BackplaneProxyEnvName)
+	// Warn user if url defined in the config file
+	if viper.GetString("url") != "" {
+		logger.Warn("URL configuration is deprecated, Pls remove backplane config URL setting")
+	}
+
+	// Fetch backplane URL from ocm env
+	bpConfig.URL, err = bpConfig.GetBackplaneURL()
 	if err != nil {
 		return bpConfig, err
 	}
 
-	bpConfig.URL = viper.GetString("url")
+	// Check if user has explicitly defined backplane URL via env; it has higher precedence over the ocm env URL
+	url, isURLViaEnv := os.LookupEnv(info.BackplaneURLEnvName)
+	if isURLViaEnv {
+		logger.Infof("backplane URL set via env vars: %s", url)
+		bpConfig.URL = url
+	}
```

Comment:
We don't need to explicitly do this, Viper does this for us automatically via `viper.AutomaticEnv()` :smiley: 

Ref: https://pkg.go.dev/github.com/spf13/viper#AutomaticEnv

Reply:
if we use `viper.BindEnv("url", info.BackplaneURLEnvName)` then it will bind to the config file `url` key which may be deleted soon .. 
We will assign it to `bpConfig.URL` for easy usage.

Reply:
Ah, that didn't cross my mind, in that case, let's port the environment assignment into a modular function so that we can reuse it for other assignments in the future.

```
func getBackplaneEnv (key string) *select_return_type* {
        val, ok := os.LookupEnv(key)
        if ok {
            logger.Infof("backplane URL set via env vars: %s", url)
           return _
        }
    }
 ```
 The return type is on you to choose, you can either return the type `BackplaneConfiguration` or a `bool` to later evaluate it.

Reply:
Thanks for the generic function suggestion, and move into private func 
```
func getBackplaneEnv(key string) (string, bool) {
}
```